components/script/dom/bindings/refcounted.rs

```diff
@@ -64,6 +64,7 @@ impl TrustedReference {
 /// in asynchronous operations. The underlying DOM object is guaranteed to live at least
 /// as long as the last outstanding `TrustedPromise` instance. These values cannot be cloned,
 /// only created from existing Rc<Promise> values.
+#[derive(Clone)]
```

Comment:
This is not safe. We need to make a new TrustedPromise value from the `Rc<Promise>` value instead.